### PR TITLE
Framework: Try out HardSourceWebpackPlugin

### PR DESCRIPTION
**Pull request by ehg**
Use [HardSourceWebpackPlugin](https://www.npmjs.com/package/hard-source-webpack-plugin) to cache built modules to disk. This cuts subsequent build time from ~60s to ~6. 

**Before**
```
Hash: e64996f437b15819a3a7
Version: webpack 1.13.1
Time: 16340ms
                    Asset     Size  Chunks             Chunk Names
    bundle-development.js  2.28 MB       0  [emitted]  main
bundle-development.js.map  2.94 MB       0  [emitted]  main
    + 992 hidden modules
wp-calypso booted in 947ms - http://calypso.localhost:3000

Getting bundles ready, hold on...
Hash: f5013d1657a7dd4dbe5e
Time: 41971ms
```

**After** (Subsequent builds)
```
Hash: 312d2ad1e6c28a59d8f3
Version: webpack 1.13.1
Time: 2243ms
                    Asset     Size  Chunks             Chunk Names
    bundle-development.js  2.19 MB       0  [emitted]  main
bundle-development.js.map  2.82 MB       0  [emitted]  main
    + 961 hidden modules
wp-calypso booted in 1015ms - http://calypso.localhost:3000

Getting bundles ready, hold on...
Hash: fcb94d6985acde5c9a4c
Time: 5276ms
```

**To Do**
- [x] update plugin version
- [x] further testing?
- [ ] ~~documentation?~~ Wait until turned on by default

/cc @seear @ockham 

Test live: https://calypso.live/?branch=try/webpack-caching

### CODE REVIEW

**Comment by blowery** [comment]
Wow, this cuts `make run` down by 35s for me locally on rebuilds. Impressive!

**Comment by seear** [comment]
> Wow, this cuts make run down by 35s for me locally on rebuilds. Impressive!

The later plugin versions do not seem to be using the cached server modules. Investigating.

File: webpack.config.js [hunk 1]
```diff
@@ -12,7 +12,8 @@ var webpack = require( 'webpack' ),
  */
 var config = require( './server/config' ),
 	sections = require( './client/sections' ),
-	ChunkFileNamePlugin = require( './server/bundler/plugin' );
+	ChunkFileNamePlugin = require( './server/bundler/plugin' ),
+	HardSourceWebpackPlugin = require('hard-source-webpack-plugin');
```
Comment (aduth) [inline webpack.config.js]:
```
wp-calypso $ ./node_modules/.bin/eslint --rule 'space-in-parens: 2' --quiet webpack.config.node.js 

/Users/andrew/Documents/Code/wp-calypso/webpack.config.node.js
  8:35  error  There must be a space inside this paren  space-in-parens
  8:64  error  There must be a space inside this paren  space-in-parens

✖ 2 problems (2 errors, 0 warnings)
```

**Comment by nylen** [comment]
From internal discussion today:

> HardSourceWebpackPlugin in its current form doesn't work for wp-calypso and requires weird cache breaking rules.  It is going to be stably integrated into webpack core as part of the webpack 5 release.  The spec is pretty interesting: https://github.com/webpack/webpack/issues/6527
> Its pretty exciting that the biggest focus for 5 is going to be performance

> Yeah, I used it for a while with the flag after merge. It would die when reading from disk.

So this should be deleted from config and package.json.

**Pull request merged**